src/components/ask-questions-page/ask-questions.js

```diff
@@ -29,30 +41,62 @@ class AskQuestions extends Component {
           "code": this.state.codeLink }
       )})
       .then(res => res.json())
-      .then(res => console.log(res))
-      //sends user to question posted page 
-      this.props.history.push('/question-posted')
+  }
+
+  showSignupModal = () => {
+    if (this.state.showSignup) {
+      return <Login close={this.toggleSignUp} signedIn={this.signedIn}/>;
     }
   }
 
+  toggleSignUp = () => {
+    this.setState({showSignup: !this.state.showSignup})
+  }
+
+  signedIn = async (token) => {
+    //sends question to database and then sends user to question posted page 
+    await this.postQuestion(token);
+    this.props.history.push('/question-posted');
+  }
+
+  componentDidMount() {
+    //populates input field on user searched term from landing page
+    if (this.props.location.state) {
+      this.setState({
+        title: this.props.location.state
+      })
+    }
+  }
+
+  componentWillUnmount() {
+    //The below componentWillUnmount is needed for warning issues with React
+    this.setState = (state,callback )=> {
```

Comment:
state and callback are supposed to just be placeholders and are never used.. i think we could remove them unless we need them

Reply:
done